app/assess/data.py

```diff
@@ -13,18 +13,26 @@
 from flask import current_app
 
 
-def get_data(endpoint: str):
+def get_data(endpoint: str, payload: Dict = None):
     if Config.USE_LOCAL_DATA:
         current_app.logger.info(f"Fetching local data from '{endpoint}'.")
-        data = get_local_data(endpoint)
+        return get_local_data(endpoint)
     else:
-        current_app.logger.info(f"Fetching data from '{endpoint}'.")
-        response = requests.get(endpoint)
+        if payload:
+            current_app.logger.info(
+                f"Fetching data from '{endpoint}', with payload: {payload}."
+            )
+            response = requests.get(endpoint, payload)
+        else:
+            current_app.logger.info(f"Fetching data from '{endpoint}'.")
+            response = requests.get(endpoint)
         if response.status_code == 200:
-            data = response.json()
+            return response.json()
         else:
+            current_app.logger.error(
+                f"Could not get data for endpoint '{endpoint}' "
+            )
             return None
```

Comment:
Wonder if it would be better to actually raise an exception that bubbles to the top so people know that's actually what's went wrong as opposed to when `None` is returned and they try to access it, just food for thought though not specific to your PR